plugins/inputs/prometheus/target_allocator.go

```diff
@@ -0,0 +1,192 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent/cfg/envconfig"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	otelpromreceiver "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+	tamanager "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/targetallocator"
+	promconfig "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/scrape"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/yaml.v3"
+)
+
+const DEFAULT_TLS_CA_FILE_PATH = "/etc/amazon-cloudwatch-observability-agent-cert/tls-ca.crt"
+
+type TargetAllocatorManager struct {
+	enabled             bool
+	manager             *tamanager.Manager
+	config              *otelpromreceiver.Config
+	host                component.Host
+	sm                  *scrape.Manager
+	dm                  *discovery.Manager
+	smLiveCh            chan struct{}
+	dmLiveCh            chan struct{}
+	taReadyCh           chan struct{}
+	reloadConfigHandler func(config *promconfig.Config)
+}
+
+func isPodNameAvailable() bool {
+	podName := os.Getenv(envconfig.PodName)
+	if podName == "" {
+		return false
+	}
+	return true
+}
+func loadConfigFromFilename(filename string) (*otelpromreceiver.Config, error) {
```

Comment:
nit: function spacing
```suggestion
}

func loadConfigFromFilename(filename string) (*otelpromreceiver.Config, error) {
```